### DISCUSSION TITLE
How to change TextEdit background color?

### DISCUSSION THREAD

**Discussion by wellitecho** [emilk/egui#1539] [Q&A] [answered]
I want to change the background color of the TextEdit based on user input. If the input does not satisfy a regex, the background will be red, otherwise it stays normal.
I looked through the docs, searched the discussions, but still couldn't figure out how to implement this.

Anyone have any idea on this?

Thanks.

**Comment by willser** [comment, marked as answer]
You could try following code,I don't know is that a best way or not,but it works.

```rust
 ui.with_layout(Layout::left_to_right().with_cross_align(Align::Min), |ui| {
    // ui.add(Label::new("REQUEST NAME: ").wrap(true));
    ui.style_mut().visuals.extreme_bg_color = Color32::RED;
    ui.text_edit_singleline(&mut self.name);
});

ui.text_edit_singleline(&mut self.url);
```
![image](https://user-images.githubusercontent.com/64425242/165767890-7f3f560c-00f1-4161-9003-e6148c9dca92.png)

**Reply by wellitecho** [reply]
Thank you, it works.
